### PR TITLE
Remove ability to reject requests, other simplifications

### PR DESCRIPTION
**Pull request by dschenkelman**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -171,30 +169,15 @@ BaaSServer.prototype._handler = function (socket) {
...
-        }
-
+      if (!this._workers.length){
+        // no available workers, queue and wait
+        this._queue.push({request, done});
         return callback();
       }
 
+      // all workers are the same, pop is faster than shift: http://stackoverflow.com/questions/6501160/why-is-pop-faster-than-shift
+      const worker = this._workers.pop();
```
Comment (jfromaniello) [inline server.js]: pop is faster than shift on arrays with more than 120k elements according to the second answer.

But I agree, a free worker is a free worker, regardless if it was the last used.

File: server.js [hunk 2]
```diff
@@ -150,16 +149,15 @@ BaaSServer.prototype._handler = function (socket) {
...
       const start = new Date();
-      const done = (worker, enqueued) => {
+      const done = (wrk, enqueued) => {
         return (err, response) => {
           log.info({
             request:    request.id,
             connection: socket._connection_id,
             took:       new Date() - start,
-            worker:     worker.id,
+            wrk: wrk.id,
```
Comment (jfromaniello) [inline server.js]: I think `worker_id` will be better.

Comment (dschenkelman) [inline server.js]: I wanted to differentiate the parameter `worker` from the one in the closure. Should we create a variable `worker_id`?

File: server.js (hunk 1 above)
Comment (dschenkelman) [inline server.js]: Yes. This is mostly my problem of shifting an array to get the first position when it is not needed, too much time at Algorithms II in FIUBA. Speed is marginal here

File: server.js (hunk 2 above)
Comment (dschenkelman) [inline server.js]: nvm, go it. change in log was unintentional

**Pull request merged**